lib/topological_inventory/providers/common/operations/async_worker.rb

```diff
@@ -0,0 +1,42 @@
+require "topological_inventory/providers/common/logging"
+require "topological_inventory/providers/common/operations/health_check"
+
+module TopologicalInventory
+  module Providers
+    module Common
+      module Operations
+        class AsyncWorker
+          include Logging
+
+          def initialize(queue, impl_class)
+            @queue = queue
+            @impl = impl_class
+          end
+
+          def run
+            Thread.new { listen }
+          end
+
+          def listen
+            loop do
+              # the queue thread waits for a message to come during `Queue#pop`
+              msg = @queue.pop
+              process_message(msg)
+            end
+          end
+
+          def process_message(msg)
+            @impl.process!(msg)
+          rescue => e
+            model, method = msg.message.to_s.split(".")
+            logger.error("#{model}##{method}: async worker failure: #{e.message}")
+          ensure
+            msg.ack
+            TopologicalInventory::Providers::Common::Operations::HealthCheck.touch_file
+            logger.debug("Operations::AsyncWorker queue length: #{@queue.length}")
```

Comment:
As debug is not a real debug in our case, I'm afraid about too much records in the Kibana. 
For example we can make reduction with some threshold timestamp (it will log maximally once per minute etc.)

Reply:
So - I changed it to this: `logger.debug(....) if queue.length >= 20 && queue.length % 5 == 0` 

that way it'll only log every 5 messages that are 20 and above, that way we can keep track of the LARGE backlogs.